frontend/simple/views/containers/contributions/IncomeForm.vue

```diff
@@ -0,0 +1,245 @@
+            <payment-methods />
+          </fieldset>
+        </div>
+
+        <group-pledges-graph class="column c-pledgesGraph"
+          :userPledgeAmount="userPledgeAmount"
+          :userIncomeAmount="userIncomeAmount"
+        />
+      </div>
+
+      <div class="field is-grouped">
+        <p class="control">
+          <i18n tag="button" class="button is-primary" type="submit">{{saveButtonText}}</i18n>
+        </p>
+        <p class="control">
+          <i18n tag="button" class="button" type="button" @click="cancelForm">Cancel</i18n>
+        </p>
+      </div>
+      </form>
+  </modal>
+</template>
+<style lang="scss" scoped>
+@import "../../../assets/sass/theme/index";
+
+.c-wrapper {
+  width: 50rem;
+  max-width: 100%;
+  height: 100%;
+  overflow: auto;
+
+  @include mobile {
+    height: 100vh;
+  }
+
+  &-columns {
+    @include mobile {
+      flex-direction: column;
+    }
+  }
+}
+
+.c-help-extra {
+  margin-top: $gi-spacer-sm;
+}
+
+.c-form,
+.c-pledgesGraph {
+  flex-basis: 50%;
+}
+
+.c-pledgesGraph {
+  align-items: flex-start;
+  flex-grow: 1;
+
+  @include mobile {
+    order: -1; // display on the top for better understanding on small screens
+    padding: $gi-spacer $gi-spacer-sm;
+  }
+
+  @include tablet {
+    justify-content: center;
+    margin-top: -$gi-spacer-lg;
+  }
+}
+
+</style>
+<script>
+import { validationMixin } from 'vuelidate'
+import { requiredIf, required, maxValue } from 'vuelidate/lib/validators'
+import InputAmount from './InputAmount.vue'
+import PaymentMethods from './PaymentMethods.vue'
+import Modal from '../../components/Modal/ModalBasic.vue'
+import TextWho from '../../components/TextWho.vue'
+import { debounce } from '../../../utils/giLodash.js'
+import GroupPledgesGraph from '../GroupPledgesGraph.vue'
+
+export default {
+  name: 'IncomeForm',
+  mixins: [ validationMixin ],
+  components: {
+    Modal,
+    InputAmount,
+    TextWho,
+    PaymentMethods,
+    GroupPledgesGraph
+  },
+  props: {
+    isEditing: Boolean
+  },
+  data () {
+    return {
+      ephemeral: {
+        formVerified: false
+      },
+      form: {
+        option: null,
+        income: null,
+        pledge: null
+      },
+      // -- Hardcoded Data just for layout purposes:
+      fakeStore: {
+        mincome: 500,
+        mincomeFormatted: '$500'
+      }
+    }
+  },
+  computed: {
+    userPledgeAmount () {
+      return this.hasMinIncome & !!this.form.pledge ? Number(this.form.pledge) : null
+    },
+    userIncomeAmount () {
+      return this.hasLessIncome & !!this.form.income ? Number(this.form.income) : null
+    },
+    maxIncome () {
+      return this.fakeStore.mincome - 1
+    },
+    infoRequired () {
+      return this.L('This information is required.')
+    },
+    hasLessIncome () {
+      return this.$v.form.option.$model === 'no'
+    },
+    hasMinIncome () {
+      return this.$v.form.option.$model === 'yes'
+    },
```

Comment:
`this.$v.form.<whatever>.$modal` shouldn't be called here in these computed functions (I don't think). According to vuelidate docs, accessing `this.$v.form.<whatever>.$modal` [is the same thing](https://monterail.github.io/vuelidate/#sub-v-values) as accessing `this.form.whatever`, and the only time the `$modal` version is useful is when assigning to `v-modal` in the HTML so that `$dirty` is automatically set when the variable is written to.

Reply:
Hmmm I don't think so. When accessing `this.$v.form.option` it returns all methods available to that input. If we need to access its value we must use `.$model`.

I added all these computed methods like `hasMinIncome` (now renamed to `canPledge`) because that verification is used in ~5 different places

Reply:
I think you misunderstood me, I was saying to access it via `this.form.option`, not `this.$v.form.option`

Reply:
Oh sorry yes I did, fixed and DRYed even more vuelidate stuff on the way